### ISSUE TITLE
False positive for missing dependencies with nested eslint config

### ISSUE THREAD

**Issue by aarongoldenthal** [depcheck/depcheck#764]
### Bug Description

I have a custom `stylelint` configuration ([here](https://www.npmjs.com/package/@aarongoldenthal/stylelint-config-standard)) with several plugins, with code in `index.js` in that package. The project also uses a custom `eslint` configuration ([here](https://www.npmjs.com/package/@aarongoldenthal/eslint-config-standard)), with configurations for `eslint`, as well as several plugins.

When including each of the `eslint` configurations in the custom package separately, there is no issue, e.g.

`.eslintrc.json`

```json
{
  "extends": [
    "@aarongoldenthal/eslint-config-standard/jest-config",
    "@aarongoldenthal/eslint-config-standard"
  ]
}
```

The package also includes a `recommended` configuration, which `extends` all of the various configurations, including the `./index.js` in that package.

`.eslintrc.json`

```json
{
  "extends": ["@aarongoldenthal/eslint-config-standard/recommended"]
}
```

And in `@aarongoldenthal/eslint-config-standard` this is:

`recommended.js`

```js
module.exports = {
    extends: [
        ...
        './index.js',
        ...
    ]
};
```

When this is used, `depcheck` resolves the `./index.js` in that configuration package as the `./index.js` in the parent package, and tries to resolve those dependencies as `eslint` plugins. It then identifies non-existent missing dependencies as eslint plugins.

```sh
PS C:\Stylelint-Config-Standard> depcheck
Missing dependencies
* eslint-plugin-stylelint-csstree-validator: .\.eslintrc.json
* eslint-plugin-stylelint-order: .\.eslintrc.json
```

These dependencies, however, do not exist. If `./index.js` is removed from the custom `eslint` configuration package the missing dependencies are not reported. Depcheck does also identify unused dependencies for the actual dependencies `stylelint-csstree-validator` and `stylelint-order`, but this is expected and they are ignored.

### Code snippets

`index.js`

```js
module.exports = {
    ...
    plugins: ['stylelint-csstree-validator', 'stylelint-order'],
    ...
};
```

Where the dependency is listed in package.json:

```
  "dependencies": {
    "stylelint-csstree-validator": "^2.0.0",
    "stylelint-order": "^5.0.0"
  },
```

### Versions

- `node -v`: 18.12.1
- `npm -v`: 8.19.2
- `depcheck --version`: 1.4.3

### Extra info

BONUS POINTS: If you wish to help us debug further, the `depcheck --json` option will show more information. Check [here in the README](https://github.com/depcheck/depcheck#example) for documentation on json output.

<details><summary>JSON</summary>

```
{
  "dependencies": [
    "postcss-html",
    "stylelint-csstree-validator",
    "stylelint-order"
  ],
  "devDependencies": [],
  "missing": {
    "eslint-plugin-stylelint-csstree-validator": [
      "C:\\Stylelint-Config-Standard\\.eslintrc.json"
    ],
    "eslint-plugin-stylelint-order": [
      "C:\\Stylelint-Config-Standard\\.eslintrc.json"
    ]
  },
  "using": {
    "@aarongoldenthal/eslint-config-standard": [
      "C:\\Stylelint-Config-Standard\\.eslintrc.json"
    ],
    "eslint-plugin-stylelint-csstree-validator": [
      "C:\\Stylelint-Config-Standard\\.eslintrc.json"
    ],
    "eslint-plugin-stylelint-order": [
      "C:\\Stylelint-Config-Standard\\.eslintrc.json"
    ],
    "eslint": [
      "C:\\Stylelint-Config-Standard\\.eslintrc.json",
      "C:\\Stylelint-Config-Standard\\package.json"
    ],
    "jest-junit": ["C:\\Stylelint-Config-Standard\\jest.config.json"],
    "stylelint-config-prettier": [
      "C:\\Stylelint-Config-Standard\\package.json"
    ],
    "jest": ["C:\\Stylelint-Config-Standard\\package.json"],
    "markdownlint-cli": ["C:\\Stylelint-Config-Standard\\package.json"],
    "prettier": ["C:\\Stylelint-Config-Standard\\package.json"],
    "stylelint": [
      "C:\\Stylelint-Config-Standard\\package.json",
      "C:\\Stylelint-Config-Standard\\tests\\index.test.js"
    ]
  },
  "invalidFiles": {},
  "invalidDirs": {}
}
```

</details>

**Comment by aarongoldenthal** [comment]
This is still an issue.

**Comment by aarongoldenthal** [comment]
This is not an issue with ESLint flat config, so closing.
